python/ray/autoscaler/v2/scheduler.py

```diff
@@ -318,7 +318,22 @@ def _try_schedule_one(self, request: ResourceRequest, is_constraint: bool) -> bo
             True if the resource request is scheduled on this node.
         """
 
-        # TODO: add labels when implementing the gang scheduling.
+        # Check if there's placement constraints that are not satisfied.
+        for constraint in request.placement_constraints:
+            if constraint.HasField("anti_affinity"):
+                anti_affinity = constraint.anti_affinity
+                if (
+                    anti_affinity.label_name in self.labels
```

Comment:
a little bit lost.. can you remind me how we convert pg into labels again?

Reply:
For a `pg=([CPU:1, CPU:1], strict_spread` we add anti-affinity constraints as `{ pg_<pg_id>: ""}`

Specifically, this 
https://github.com/ray-project/ray/blob/55838fc47bf5531d7c73c9b48064a6d1fd4f2d12/src/ray/gcs/gcs_server/gcs_autoscaler_state_manager.cc#L145-L171

and this:

https://github.com/ray-project/ray/blob/55838fc47bf5531d7c73c9b48064a6d1fd4f2d12/src/ray/gcs/pb_util.h#L408-L429